projects/plugins/jetpack/extensions/blocks/sharing-button/sharing-sources-block.php

```diff
@@ -0,0 +1,1397 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+/**
+ * Define all sharing sources.
+ *
+ * phpcs:disable Generic.Files.OneObjectStructurePerFile.MultipleFound
+ */
+
+namespace Automattic\Jetpack\Extensions\Sharing_Button_Block;
+
+use Automattic\Jetpack\Device_Detection\User_Agent_Info;
+use Jetpack_PostImages;
+
+/**
+ * Base class for sharing sources.
+ * See individual sharing classes below for the implementation of this class.
+ */
+abstract class Sharing_Source_Block {
+	/**
+	 * Button style (icon, icon-text, text, or official).
+	 *
+	 * @var string
+	 */
+	public $button_style;
+
+	/**
+	 * Does the service have an official version.
+	 *
+	 * @var bool
+	 */
+	public $smart;
+
+	/**
+	 * Should the sharing link open in a new tab.
+	 *
+	 * @var bool
+	 */
+	protected $open_link_in_new;
+
+	/**
+	 * Sharing unique ID.
+	 *
+	 * @var int
+	 */
+	protected $id;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param int $id       Sharing source ID.
+	 */
+	public function __construct( $id ) {
+		$this->id = $id;
+		/**
+		 * Filter the way sharing links open.
+		 *
+		 * By default, sharing links open in a new window.
+		 *
+		 * @module sharedaddy
+		 *
+		 * @since 3.4.0
+		 *
+		 * @param bool true Should Sharing links open in a new window. Default to true.
+		 */
```

Comment:
Since all the filters are already defined in the module, you should be able to remove all the docblocks, and replace them with `/** This filter is documented in projects/plugins/jetpack/modules/sharedaddy/sharing-sources.php */`

Reply:
Hmm, I thought we would eventually remove the module? Or we will still keep it?

Reply:
We will have to keep it around for a little while, we cannot remove it as soon as the block will be available.

Reply:
I'm still wondering on how many filters we would actually need at the end, so I tried not to delete anything might be useful. I could remove notes on module though, as this block should work on it's own.